DirectProgramming/C++SYCL_FPGA/Tutorials/Features/IPAuthoring_interfaces_overview/csr-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+
+// oneAPI headers
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/experimental/pipes.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAdd_pipes;
+
+// Forward declare pipe names to reduce name mangling (maybe get rid of; pipes
+// already have bad names)
+class ID_PipeA;
+class ID_PipeB;
+
+#define VECT_SIZE 256
+
+using PipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>));
+
+using InputPipeA =
+    sycl::ext::intel::experimental::pipe<ID_PipeA, int, 0,
+                                         PipeProps>;
+using InputPipeB =
+    sycl::ext::intel::experimental::pipe<ID_PipeB, int, 0,
+                                         PipeProps>;
+
+using CSRPipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::protocol_avalon_mm_uses_ready));
```

Comment:
```suggestion
using CSRPipeProps = decltype(sycl::ext::oneapi::experimental::properties(
    sycl::ext::intel::experimental::protocol_avalon_mm));
```
Rob said this bug was fixed in 2024.0